tests/core/test_accounts.py

```diff
@@ -362,24 +401,56 @@ def test_eth_account_recover_transaction_from_eth_test(acct, transaction):
     assert acct.recoverTransaction(raw_txn) == expected_sender
 
 
-def test_eth_account_encrypt(acct, web3js_key, web3js_password):
-    encrypted = acct.encrypt(web3js_key, web3js_password)
+@pytest.mark.parametrize(
+    'private_key, password, expected_decrypted_key',
+    [
+        (
+            web3js_key(),
+            web3js_password(),
+            to_bytes(hexstr=web3js_key()),
+        ),
+        (
+            web3js_private_key(web3js_key()),
+            web3js_password(),
+            web3js_private_key(web3js_key()),
+        ),
+    ],
+    ids=['hex_str', 'eth_keys.datatypes.PrivateKey']
+)
+def test_eth_account_encrypt(acct, private_key, password, expected_decrypted_key):
+    encrypted = acct.encrypt(private_key, password)
 
     assert encrypted['address'] == '2c7536e3605d9c16a7a3d7b1898e529396a65c23'
     assert encrypted['version'] == 3
 
-    decrypted_key = acct.decrypt(encrypted, web3js_password)
+    decrypted_key = acct.decrypt(encrypted, password)
 
-    assert decrypted_key == to_bytes(hexstr=web3js_key)
+    assert decrypted_key == expected_decrypted_key
 
 
-def test_eth_account_prepared_encrypt(acct, web3js_key, web3js_password):
-    account = acct.privateKeyToAccount(web3js_key)
-    encrypted = account.encrypt(web3js_password)
+@pytest.mark.parametrize(
+    'private_key, password, expected_decrypted_key',
+    [
+        (
+            web3js_key(),
+            web3js_password(),
+            to_bytes(hexstr=web3js_key()),
+        ),
+        (
+            web3js_private_key(web3js_key()),
+            web3js_password(),
+            web3js_private_key(web3js_key()),
```

Comment:
Same here, should by `HexBytes` type.

Reply:
Also:
```
(venv) [06:38:54 PM] @stefanmendoza ➜  eth-account git:(issue-22) ✗ python
Python 3.6.5 (default, Jun 17 2018, 12:13:06) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from eth_keys import keys
>>> from hexbytes import HexBytes
>>> pkey = keys.PrivateKey(HexBytes('0x4c0883a69102937d6231471b5dbb6204fe5129617082792ae468d01a3f362318'))
>>> type(pkey)
<class 'eth_keys.datatypes.PrivateKey'>
>>> type(pkey.to_bytes())
<class 'hexbytes.main.HexBytes'>
>>> pkey == pkey.to_bytes()
True
```

Reply:
```
>>> pkey == pkey.to_bytes()
True
```

I had a feeling that ^ might be true, but didn't remember without looking it up.

Just for clarity in the test, I think we should assert the type in the test, since it's made explicit in the docs. Also, let's use an expected result that is explicitly bytes (rather than a `PrivateKey` that happens to `==` bytes).

Reply:
Do we want to force the type to be `HexBytes` here?
```
return HexBytes(decode_keyfile_json(keyfile, password_bytes))
```
Or would we rather just leave it as
```
return decode_keyfile_json(keyfile, password_bytes)
```
and flex on the `bytes` type?

Reply:
Let's do the first: force it to `HexBytes`. Obviously this type bug predates the PR. Thanks for helping ID and fix it! :)

Reply:
Fixed here: https://github.com/ethereum/eth-account/pull/30/commits/b501da3a3ba80fb9e62ce58b20575ca91f2548e7#diff-1de692d823115945c2232e1aa3066811